features/accommodations/creation/AccommodationMultiStepForm.tsx

```diff
@@ -0,0 +1,158 @@
+    bookBeforeTime: 0,
+    minNights: 2,
+    maxNights: 7,
+    price: 1,
+    isPerGuest: true,
+    hostId: "11111111-1111-1111-1111-111111111111",
+    houseRules: "",
+    propertyTypeId: "",
+    location: {
+      id: "",
+      country: "",
+      region: "",
+      city: "",
+      postalCode: "",
+      address: "",
+      apt: "",
+      latitude: 0,
+      longitude: 0,
+    },
+    amenities: [],
+    availabilities: [],
+    priceDiffs: [],
+  });
+  const [selectedFiles, setSelectedFiles] = useState<any>();
+  const [imagePreviews, setImagePreviews] = useState<ImageOrder[]>([]);
+
+  const { propertyTypes } = usePropertyTypes();
+  const { amenities } = useAmenities();
+
+  function updateFields(fields: Partial<CreateAccommodation>) {
+    setData((prev: any) => ({ ...prev, ...fields }));
+  }
+  const MapWithNoSSR = dynamic(() => import("./MapForm"), {
+    ssr: false,
+  });
+  const { steps, currentStepIndex, step, isFirstStep, isLastStep, back, next } = useMultistepForm([
+    <PropertyForm
+      key="Property"
+      allPropertyTypes={propertyTypes}
+      updateFields={updateFields}
+      {...data}
+    />,
+    <RoomsForm key="Rooms" updateFields={updateFields} {...data} />,
+    <LocationForm key="Location" updateFields={updateFields} {...data} />,
+    <MapWithNoSSR key="Map" updateFields={updateFields} {...data} />,
+    <AmenitiesForm
+      key="Amenities"
+      allAmenities={amenities}
+      updateFields={updateFields}
+      {...data}
+    />,
+    <TitleAndDescriptionForm key="Title and description" updateFields={updateFields} {...data} />,
+    <PhotosForm
+      key="Photos"
+      imagePreviews={imagePreviews}
+      updateFiles={setSelectedFiles}
+      updateImagePreviews={setImagePreviews}
+    />,
+    <AvailabilityForm key="Availability" />,
+    <PricingForm key="Pricing" />,
+    <HouseRulesForm key="House rules" />,
+    <FinishForm key="Finish" />,
+  ]);
+
+  const notifications = useNotifications();
+  const { setResult } = useResult("accommodations");
+
+  const createAccommodationAction = useAction<CreateAccommodation>(createAccommodation, {
+    onSuccess: (a) => {
+      notifications.success("You have successfully created a new accommodation.");
+      setResult({ status: ResultStatus.Ok, type: "CREATE_ACCOMMODATION" });
+      console.log(a);
+      uploadFiles(
+        selectedFiles,
+        imagePreviews.map((io) => io.index),
+        a.id
+      )
+        .then(() => notifications.success("You have successfully uploaded photos."))
+        .catch(() => notifications.error("Error while uploading photos."));
+    },
+    onError: (error: any) => {
+      notifications.error(extractErrorMessage(error));
+      setResult({ status: ResultStatus.Error, type: "CREATE_ACCOMMODATION" });
+    },
+  });
+
+  function onSubmit(e: FormEvent) {
+    e.preventDefault();
+    // console.log(data);
+    if (!isLastStep) return next();
+    createAccommodationAction(data);
+  }
+
+  return (
```

Comment:
Is there any other case where we would need a multistep form?

Reply:
Maybe on the reservation page. But even if we don't really need it, I would extract it because I think it is super cool and we can reuse it on some other project.